src/sourmash/tax/__main__.py

```diff
@@ -299,36 +300,68 @@ def annotate(args):
         error(f'ERROR: No taxonomic assignments loaded from {",".join(args.taxonomy_csv)}. Exiting.')
         sys.exit(-1)
 
-    # get gather_csvs from args
-    gather_csvs = tax_utils.collect_gather_csvs(args.gather_csv, from_file=args.from_file)
+    # get csv from args
+    input_csvs = tax_utils.collect_gather_csvs(args.gather_csv, from_file=args.from_file)
 
     # handle each gather csv separately
-    for n, g_csv in enumerate(gather_csvs):
-        query_gather_results = tax_utils.check_and_load_gather_csvs(gather_csvs, tax_assign, force=args.force,
-                                                                                       fail_on_missing_taxonomy=args.fail_on_missing_taxonomy,
-                                                                                       keep_full_identifiers=args.keep_full_identifiers,
-                                                                                       keep_identifier_versions = args.keep_identifier_versions,
-                                                                                       lins=args.lins)
-
-        if not query_gather_results:
-            continue
-
-        out_base = os.path.basename(g_csv.rsplit('.csv')[0])
-        this_outfile, limit_float = make_outfile(out_base, "annotate", output_dir=args.output_dir)
-
-        header = [field.name for field in fields(GatherRow)]
-        with FileOutputCSV(this_outfile) as out_fp:
-            header.append("lineage")
-            w = csv.DictWriter(out_fp, header, delimiter=',')
-            w.writeheader()
-
-            for gather_res in query_gather_results:
-                for taxres in gather_res.raw_taxresults:
-                    gr = asdict(taxres.raw)
-                    write_gr = {key: gr[key] for key in gr if key in header}
-                    write_gr['lineage'] = taxres.lineageInfo.display_lineage(truncate_empty=True)
-                    w.writerow(write_gr)
+    for n, in_csv in enumerate(input_csvs):
```

Comment:
could you make a new issue to refactor this section? I dislike really long try/except blocks but that's not a blocker for this PR (I think it's beyond scope).

Reply:
Sure -- would moving the internal annotation code to a function solve that? Or is there a different refactor you have in mind?

The try/except is just saving me from having to check the 'force' arg for each potential ValueError.

Reply:
file the issue and find out ;).

(I'm thinking this is a place where plugins could be potentially used.)

Reply:
Hah, ok! I was just going to do the refactor here if easy, but I will file an issue instead 😆

Reply:
see #2520